civis/ml/tests/test_model.py

```diff
@@ -173,7 +173,7 @@ def test_stash_local_data_from_dataframe_csv(mock_file):
     assert _model._stash_dataframe_as_csv(df, mock.Mock()) == -11
     mock_file.assert_called_once_with(mock.ANY, name='modelpipeline_data.csv',
                                       client=mock.ANY)
-    assert isinstance(mock_file.call_args[0][0], BytesIO)
+    assert isinstance(mock_file.call_args[0][0], StringIO)
```

Comment:
Shouldn't this fail in the Python27 tests?

Reply:
Good question! Behold --

Python 2.7, six 1.11.0:

```python
In [2]: from six import StringIO, BytesIO

In [3]: s = StringIO()

In [4]: b = BytesIO()

In [5]: isinstance(s, StringIO)
Out[5]: True

In [6]: isinstance(s, BytesIO)  # This is why we didn't get the expected failure?
Out[6]: True

In [7]: isinstance(b, StringIO)
Out[7]: True

In [8]: isinstance(b, BytesIO)
Out[8]: True
```

Python 3.6, six 1.11.0 (where everything makes sense to me):

```python
In [1]: from six import StringIO, BytesIO

In [2]: s = StringIO()

In [3]: b = BytesIO()

In [4]: isinstance(s, StringIO)
Out[4]: True

In [5]: isinstance(s, BytesIO)
Out[5]: False

In [6]: isinstance(b, StringIO)
Out[6]: False

In [7]: isinstance(b, BytesIO)
Out[7]: True
```

Reply:
Huh.  Thanks for looking into that.

Reply:
I think the test passes because of https://github.com/benjaminp/six/blob/1.11.0/six.py#L664 . Given that equality, do we need the `if six.PY3` conditional?

Reply:
Good point! Although I don't have a problem merging as is, either.